fuel_gauge: restructure cmake libraries

matt-wood-ct opened the pull request:
Adjusted drivers to include sources into a single library inline with other drivers cmake structure

matt-wood-ct commented:
Closes #93834

teburd reviewed (dismissed):
Can you provide some reasoning behind this change, as some areas of Zephyr have chosen this direction (sensors) while others (e.g. i2c/spi) do not. But this seems mostly tied to how the directory structure is setup than any specific decision making.

What are the benefits of going this route over the existing structure? This reasoning should also go into the commit message (why is this change being made, what is being changed).

pdgendt reviewed (approved):
I agree with the submitter, this is an improvement for using `zephyr_library_amend()` with out-of-tree `fuel_gauge` drivers.

nordicjm commented:
> re-assigning to a build system person, my -1 may be dismissed once either @nordicjm or @tejlmand have chimed in but one of these folks should take a look and comment before this gets merged imho

It should be part of one `zephyr_library()` for out of tree usage. If we build a sensor sample with shell and sensor shell enabled, we get the following static library files that zephyr pulls in:
```
./qdec/app/libapp.a
./qdec/modules/hal_nordic/nrfx/libmodules__hal_nordic__nrfx.a
./qdec/zephyr/libzephyr.a
./qdec/zephyr/kernel/libkernel.a
./qdec/zephyr/drivers/timer/libdrivers__timer.a
./qdec/zephyr/drivers/serial/libdrivers__serial.a
./qdec/zephyr/drivers/sensor/libdrivers__sensor.a
./qdec/zephyr/drivers/sensor/nordic/qdec_nrfx/libdrivers__sensor__nordic__qdec_nrfx.a
./qdec/zephyr/drivers/sensor/nordic/temp/libdrivers__sensor__nordic__temp.a
./qdec/zephyr/drivers/pinctrl/libdrivers__pinctrl.a
./qdec/zephyr/drivers/gpio/libdrivers__gpio.a
./qdec/zephyr/drivers/console/libdrivers__console.a
./qdec/zephyr/drivers/clock_control/libdrivers__clock_control.a
./qdec/zephyr/subsys/rtio/libsubsys__rtio.a
./qdec/zephyr/soc/soc/nrf52840/libsoc__nordic.a
./qdec/zephyr/lib/posix/options/liblib__posix__options.a
./qdec/zephyr/lib/libc/common/liblib__libc__common.a
./qdec/zephyr/lib/libc/picolibc/liblib__libc__picolibc.a
./qdec/zephyr/arch/arch/arm/core/libarch__arm__core.a
./qdec/zephyr/arch/arch/arm/core/mpu/libarch__arm__core__mpu.a
./qdec/zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a
./qdec/zephyr/arch/common/libarch__common.a
./qdec/zephyr/arch/common/libisr_tables.a
```
Can see that timer/serial/gpio/pinctrl/console/clock_control are one file each (no matter how many drivers are included), whereas sensor is 3 different files. Now for sensor it's actually coincidentally partially working because of the library created (which it adds the shell files to) here: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/CMakeLists.txt#L78 so out of tree users wouldn't notice, unless they inspected the files and saw that in-tree sensor libraries all have distinct outputs whilst all out of tree sensors would go into the shell file :thinking:

teburd reviewed (changes requested):
Commit message needs clarification on reasoning still, a bisect or git log won't help understand why this change was done currently

matt-wood-ct committed:
fuel_gauge: restructure cmake libraries

Adjusted drivers to include sources into a single library
to allow for the use of zephyr_library_amend when extending
with out of tree drivers in workspace projects inline with other
driver groups structure

Signed-off-by: Matt Wood <matt.wood@corintech.com>

matt-wood-ct commented:
@teburd amended the commit body to add commentary about `zephyr_library_amend`, please let me know if this is acceptable.
